hackathon/cmd/updater/main.go

```diff
@@ -225,5 +232,69 @@ func main() {
 		img = hackimage.Gray(i)
 	}
 
+	if size := opt.tileSize; size > 0 {
+		tiles := hackimage.Tiles(img, size)
+		mapping := mapTiles(tiles, '0')
+		pattern, err := readPattern(opt.tilePattern)
+		if err != nil {
+			logrus.Fatalf("readPattern(%q): %v", opt.tilePattern, err)
+		}
+		img = renderPattern(mapping, size, pattern)
+	}
+	hackimage.Print(img)
 	hackupdater.Update(ctx, opt.creds, opt.confirm, convert(img), nil, opt.config, opt.group)
+
+}
+
+func mapTiles(tiles []image.Gray, ch rune) map[rune]image.Gray {
+	mapping := make(map[rune]image.Gray, len(tiles))
+	for _, t := range tiles {
+		mapping[ch] = t
+		ch++
+	}
+	return mapping
+}
+
+// http://www.asciitable.com/
+func readPattern(path string) ([][]rune, error) {
+	buf, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("read: %w", err)
+	}
+	lines := strings.Split(strings.TrimSpace(string(buf)), "\n")
+	runes := make([][]rune, len(lines))
+	for row, col := range lines {
+		runes[row] = make([]rune, len(col))
+		for c, ch := range col {
+			runes[row][c] = ch
+		}
+	}
+	return runes, nil
+}
+
+func renderPattern(tileset map[rune]image.Gray, size int, rows [][]rune) image.Gray {
+	var width int
+	for _, col := range rows {
+		if n := len(col); n > width {
+			width = n
+		}
+	}
+
+	rect := image.Rect(0, 0, width*size, size*len(rows))
+	img := image.NewGray(rect)
+	for row, cols := range rows {
```

Comment:
'cols` being an element of `rows` feels a bit strange

Reply:
cells then? Will fix next.